### ISSUE TITLE
UnicodeEncodeError Exception when task name has accented chars

### ISSUE THREAD

**Issue by pachi** [ansible/ansible#6817]
##### Summary:

When I use accented chars in task names in a playbook I get an UnicodeEncodeError Exception and ansible stops processing the playbook.
If I remove accented chars the playbook works fine.
##### Steps To Reproduce:

Create a playbook with something like:

```
- name: Creación de usuario sisgen
  user: sisgen state=present
```
##### Expected Results:

Playbook works.
##### Actual Results:

The following traceback when processing that task (previous tasks without accents work fine):

> Traceback (most recent call last):
>  File "/usr/local/bin/ansible-playbook", line 307, in <module>
> 
> sys.exit(main(sys.argv[1:]))
>  File "/usr/local/bin/ansible-playbook", line 247, in main
> 
> pb.run()
>  File "/usr/local/lib/python2.7/dist-packages/ansible/playbook/**init**.py", line 286, in run
> 
> if not self._run_play(play):
>  File "/usr/local/lib/python2.7/dist-packages/ansible/playbook/__init__.py", line 659, in _run_play
> 
> if not self._run_task(play, task, False):
>  File "/usr/local/lib/python2.7/dist-packages/ansible/playbook/__init__.py", line 405, in _run_task
> 
> self.callbacks.on_task_start(template(play.basedir, name, task.module_vars, lookup_fatal=False, filter_fatal=False), is_handler)
>  File "/usr/local/lib/python2.7/dist-packages/ansible/utils/template.py", line 321, in template
> 
> varname = template_from_string(basedir, varname, vars, fail_on_undefined)
>  File "/usr/local/lib/python2.7/dist-packages/ansible/utils/template.py", line 543, in template_from_string
> 
> data = data.decode('utf-8')
>  File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
> 
> return codecs.utf_8_decode(input, errors, True)
> UnicodeEncodeError
> : 
> 'ascii' codec can't encode character u'\xfa' in position 25: ordinal not in range(128)

I'm using ansible 1.5.4 installed from Pypi using pip.

**Comment by jimi-c** [comment]
I am unable to reproduce this, and have tested all the way back to version 1.3.0.

You note the environment as Ubuntu 13.10, but is that the OS of the system running ansible, or that of the system being managed? What is the python version of the system on which you're running anisble?

**Comment by pachi** [comment]
The OS running ansible and the system being managed (a VM created using Vagrant) is in both cases Ubuntu 13.10.
The Python version is 2.7.5+ (default on Ubuntu 13.10).

**Comment by jimi-c** [comment]
I've tried both remote and local with 13.10, and still can't reproduce this using the string you posted above. My test playbook:

<pre>
---
- hosts: all
  gather_facts: yes
  tasks:
  - name: Creación de usuario sisgen
    shell: echo "I am on $(hostname -s)"
</pre>

produces:

<pre>
# ansible --version
ansible 1.5.4 (release1.5.4 f1f63829ef) last updated 2014/04/03 07:29:20 (GMT -400)
# ansible-playbook 6817.yml -i localhost, -c local -vv
PLAY [all] ******************************************************************** 
GATHERING FACTS *************************************************************** 
<localhost> REMOTE_MODULE setup
ok: [localhost]
TASK: [Creación de usuario sisgen] ******************************************** 
<localhost> REMOTE_MODULE command echo "I am on $(hostname -s)" #USE_SHELL
changed: [localhost] => {"changed": true, "cmd": "echo \"I am on $(hostname -s)\" ", "delta": "0:00:00.001477", "end": "2014-04-03 07:31:29.119479", "rc": 0, "start": "2014-04-03 07:31:29.118002", "stderr": "", "stdout": "I am on ubuntu1310"}
PLAY RECAP ******************************************************************** 
localhost                  : ok=2    changed=1    unreachable=0    failed=0   
</pre>

Do you have any other samples that might be used to reproduce this? I have also verified with both dash and bash to make sure it's not a shell issue (ie. LANG or something else).

**Comment by pachi** [comment]
I've tried to reproduce it, and I made a bad assumption about the simplest case to trigger the problem. I happens that the bug only shows when the string has template variables, such as in:

```
name: Creación de usuario  {{ my_user }}
    user: {{ my_user }} state=present
```

In other cases I can use accented chars without problem.

Now I've noticed that the traceback fails after working with the ansible/utils/template.py module, so maybe that's where the problem is.

```
Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 307, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/usr/local/bin/ansible-playbook", line 247, in main
    pb.run()
  File "/usr/local/lib/python2.7/dist-packages/ansible/playbook/__init__.py", line 286, in run
    if not self._run_play(play):
  File "/usr/local/lib/python2.7/dist-packages/ansible/playbook/__init__.py", line 659, in _run_play
    if not self._run_task(play, task, False):
  File "/usr/local/lib/python2.7/dist-packages/ansible/playbook/__init__.py", line 405, in _run_task
    self.callbacks.on_task_start(template(play.basedir, name, task.module_vars, lookup_fatal=False, filter_fatal=False), is_handler)
  File "/usr/local/lib/python2.7/dist-packages/ansible/utils/template.py", line 321, in template
    varname = template_from_string(basedir, varname, vars, fail_on_undefined)
  File "/usr/local/lib/python2.7/dist-packages/ansible/utils/template.py", line 543, in template_from_string
    data = data.decode('utf-8')
  File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode

return codecs.utf_8_decode(input, errors, True)
UnicodeEncodeError
: 
'ascii' codec can't encode character u'\xfa' in position 25: ordinal not in range(128)
```

Hope this helps and sorry for the confusing bug report.

**Comment by jimi-c** [comment]
Ahh yes, that helps, I'll look into that.

**Comment by jimi-c** [comment]
It appears that @jctanner fixed this issue a few weeks ago with ad70e9bcd633e9c088d912d81ef97d56c696e62a, so I'm going to go ahead and close this. This fix will be included in the next major release. If you have any further questions regarding this, please let us know. Thanks!

**Comment by pachi** [comment]
Very nice, thanks a lot for your help!
